python/cog/server/http.py

```diff
@@ -88,16 +89,42 @@ def create_app(
 
     predictor_ref = get_predictor_ref(config, mode)
 
+    try:
+        # TODO: avoid loading predictor code in this process
+        predictor = load_predictor_from_ref(predictor_ref)
+        InputType = get_input_type(predictor)
+        OutputType = get_output_type(predictor)
+    except Exception as ex:
+        app.state.health = Health.SETUP_FAILED
+        response = schema.PredictionResponse(input={},
```

Comment:
I don't think this is a change you introduced, but it seems quite wrong to me that`setup` returns a `PredictionResponse`. It looks like that was introduced [here](https://github.com/replicate/cog/commit/78caaa3e27201cfc87309b7a5f62532e7eb6df72#diff-136cc6506e8aaea24081862f89e1cb778251b7726180fe237c8aaf560ae7ec69R310-R322) and I think I'd like to back that change out.